smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -225,6 +230,7 @@ public KafkaSource(Vertx vertx,
                     }
                 })
                 .map(rec -> commitHandler.received(new IncomingKafkaRecord<>(rec, commitHandler, failureHandler)))
+                .onItem().invoke(this::incomingTrace)
```

Comment:
let's build a conditional pipeline instead of testing if tracing is enabled every time.

```
multi = ...
if (getTracingEnabled()) {
    multi = .... 
}

Reply:
Ok

Reply:
How do I do the `onItem()` so that it's invoked at the correct place in the stream?

Do you mean creating two pipelines, one with and one without the `onItem()`? Or do you mean multi into multi?

Reply:
Pipelines are "append-only".

You create one, and append the `onItem()` if tracing is enabled:

```java
multi = ...
if (getTracingEnabled()) {
    multi = multi.onItem().invoke(this::incomingTrace); 
}
multi = multi....
```